### PR TITLE
developed API for created a team

### PR DESCRIPTION
**Pull request by nishadipri**
# Description

developed API for created team
Fixes # (issue)

# How to test?

Please provide a short summary how your changes can be tested?

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [ ] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [ ] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by naresh699 — COMMENTED** [review]
overall code is good, however i will suggest read about commenting on code
https://jsdoc.app/, or look for some you tube videos

File: packages/server/migrations/20240330125917_make_teamName_teamCode_unique.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+
+exports.up = function (knex) {
```
Comment (DanJecu) [inline packages/server/migrations/20240330125917_make_teamName_teamCode_unique.js:6]: You could do it this way as well ⬇️
```suggestion
exports.up = async knex => {
    await knex.schema.alterTable('Teams', table => {
    table.string('team_name').unique().alter()
    table.string('team_code').unique().alter()
    })
}
```

File: packages/server/migrations/20240330125917_make_teamName_teamCode_unique.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  return knex.schema.alterTable('Teams', async () => {
+    await knex.raw(`ALTER TABLE Teams ADD UNIQUE (team_name);`);
+    await knex.raw(`ALTER TABLE Teams ADD UNIQUE (team_code);`);
+  });
+};
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.down = function () {};
```
Comment (DanJecu) [inline packages/server/migrations/20240330125917_make_teamName_teamCode_unique.js:16]: Will be suggested to add a down function as well

```suggestion
exports.down = async knex => {
  await knex.schema.alterTable('lenusBenefit', table => {
     table.string('team_name').alter()
     table.string('team_code').alter()
  })
}

```

File: packages/server/migrations/20240321154545_create_employees_table.js [hunk 3]
```diff
@@ -4,10 +4,10 @@
  */
 exports.up = function (knex) {
   return knex.schema.createTable('Employees', (table) => {
-    table.increments('id').primary();
+    table.increments('id').primary().unique();
```
Comment (DanJecu) [inline packages/server/migrations/20240321154545_create_employees_table.js:7]: Same here, if you will create new migration files to update the schema, you will not need these changes.

**Review by DanJecu — CHANGES_REQUESTED** [review]

**Review by DanJecu — APPROVED** [review]

File: packages/server/migrations/20240321154351_create_teams_table.js [hunk 4]
```diff
@@ -4,7 +4,7 @@
  */
 exports.up = function (knex) {
   return knex.schema.createTable('Teams', (table) => {
-    table.increments('id').primary();
+    table.increments('id').primary().unique();
```
[the remark this answers is not in the document]
Comment (nishadipri) [reply packages/server/migrations/20240321154351_create_teams_table.js]: done

**Pull request merged**